src/foam/nanos/http/ThreadsWebAgent.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.nanos.http;
+
+import foam.core.*;
+import java.lang.Thread;
+import java.lang.StackTraceElement;
+import java.lang.String;
+import java.util.Set;
+import java.io.PrintWriter;
+import javax.servlet.http.HttpServletRequest;
+
+public class ThreadsWebAgent
+  implements WebAgent
+{
+  public ThreadsWebAgent() {}
+
+  public void execute(X x) {
+    final PrintWriter out  = (PrintWriter) x.get(PrintWriter.class);
+    HttpServletRequest req = (HttpServletRequest) x.get(HttpServletRequest.class);
+
+    Set<Thread> threadSet = Thread.getAllStackTraces().keySet();
+
+    Thread[] threadArray = threadSet.toArray(new Thread[threadSet.size()]);
+    out.println("<HEAD><TITLE>Threads</TITLE></HEAD>\n");
+    out.println("<pre>");
+    out.println("<H1>Threads</H1>\n");
+
+    for (Thread thread : threadArray){
```

Comment:
Follow https://github.com/foam-framework/foam2/blob/master/doc/guides/StyleGuide.md for FOAM code.